### PR TITLE
[API] Add unit tests for spam filtering in review submission

### LINKED ISSUE #823: [API] Add unit tests for spam filtering in review submission
We recently implemented backend spam filtering in the ReviewCreateView to detect and block inappropriate content using a utility function` is_spam()`. This issue focuses on verifying that logic through unit tests.

The unit tests should cover the following behaviors:

- Normal review content is accepted and saved.
- Reviews containing banned phrases (e.g., “buy now”, “free money”) are blocked.
- Reviews with excessive repeated punctuation (e.g., “!!!!!!”) are blocked.
- Reviews that are too short or repetitive (e.g., “ab”, “hi hi hi”) are flagged as spam.

The test class SpamFilterTests has been created in test_api.py and contains individual tests to verify each scenario. We will try to detect edge cases for this.

### PR DESCRIPTION
**Pull request by chrislowzhengxi**
This PR adds unit tests to validate the backend spam filtering logic in the ReviewCreateView. The spam detection logic is implemented using the is_spam() utility function and blocks inappropriate or low-quality content.

The tests cover the following scenarios:

- Normal review content is accepted and saved.
- Reviews containing banned phrases (e.g., “buy now”, “free money”) are blocked.
- Reviews with excessive repeated punctuation (e.g., “!!!!!!!!”) are blocked.
- Reviews that are too short or repetitive (e.g., “ab”, “hi hi hi”) are flagged as spam and rejected.
The test cases are implemented in the SpamFilterTests class in test_api.py, and each scenario is tested independently.

Closes: #823

### CODE REVIEW

**Comment by chrislowzhengxi** [comment]
Just modified MessageView in views.py so it passed all the tests in pytest. Ready for review. 


To test this I ran: 
`python manage.py test chigame.api.tests.test_api.SpamFilterTests`

**Review by fredricngo — APPROVED** [review]
LGTM! Ran the commands in my terminal and everything seems to be working.

**Review by MuyanXie — COMMENTED** [review]
I can see there are only four lines of change in this PR -- did we add unit test in a separate PR?

**Comment by chrislowzhengxi** [comment]
> I can see there are only four lines of change in this PR -- did we add unit test in a separate PR?

Hey Sam, just a quick heads up, the spam filter tests are definitely in the file (**SpamFilterTests is right below line 752 til the end**), but I'm not sure if it's being recognized as a separate commit or if it got bundled into the previous ones. I tried resetting and re-adding the tests manually, but Git’s being a pain and it’s not showing up clearly in the commit history. The functionality of the test is all there, but not sure if it got merged before. Thanks!

**Review by MuyanXie — APPROVED** [review]
LGTM. Thanks!

**Comment by chrislowzhengxi** [comment]
@MuyanXie I just resolved some merge conflicts here. Should be ready to merge.

File: src/chigame/api/tests/test_api.py [hunk 1]
```diff
@@ -747,85 +747,120 @@ def test_patch_lobby_unauthorized(self):
         self.assertEqual(unchanged_lobby.name, lobby.name)
 
 
-class AccessControlTests(APITestCase):
+class SpamFilterTests(APITestCase):
     def setUp(self):
```
Comment (MuyanXie) [inline src/chigame/api/tests/test_api.py:835]: what's going on here? why do we change name?

File: src/chigame/api/tests/test_api.py [hunk 2]
```diff
@@ -747,85 +747,120 @@ def test_patch_lobby_unauthorized(self):
...
+    #     self.assertIn("spam", str(response.data).lower())
+    #     self.assertEqual(Message.objects.count(), 0)
+
+    def test_review_allows_normal_content(self):
+        url = reverse("api-game-review-create", args=[self.game.id])
+
+        print("Game ID:", self.game.id)
+        from chigame.games.models import Game  # add this import at the top if needed
+
+        print("Game exists:", Game.objects.filter(id=self.game.id).exists())
```
Comment (MuyanXie) [inline src/chigame/api/tests/test_api.py:877]: also the unit tests are now too mixed up in changes. Can you explain why do you do this? I suspect this happens when you resolve merge conflicts

**Review by MuyanXie — CHANGES_REQUESTED** [review]

File: src/chigame/api/tests/test_api.py (hunk 2 above)
Comment (chrislowzhengxi) [reply src/chigame/api/tests/test_api.py:877]: I assume it's because I removed the authentication and only kept the spam tests. But then I just realized that the authentication tests are merged. Should I restore them?

**Review by MuyanXie — APPROVED** [review]
The changes looks to me and the commit is very neat and clean. Thanks! Approved

**Comment by chrislowzhengxi** [comment]
Closing: 
This PR added comprehensive unit tests to ensure the backend spam filtering logic functions as intended. Using the `SpamFilterTests` suite, we verified that inappropriate review content, such as banned keywords, punctuation abuse, and short/repetitive patterns, is correctly blocked. All tests passed, and the logic aligns with the production behavior of the `is_spam()` utility. This PR closes #823.

**Comment by MuyanXie** [comment]
Issue Score: excellent

Comments:
Nice job! Even though the PR has some mixed commits in it, the result is still very clean. Please make sure your PR is readable and structured next time
